Adding a new ACL rule 'core.options'

Bakual opened the pull request:
### Issue

As explained in issue https://github.com/joomla/joomla-cms/issues/4954, currently it's not possible to give someone access to the options without giving him also access to the permissions. It means that a user with access to the options is a "SuperUser" for that particular extension.
There may be use cases where you want to give someone access to the options without giving him full permissions for the extension.
### Solution

This PR adds a new ACL rule `core.options` which will allow to edit the options, but not the permissions.
I only did it for com_content so far, but if the concept is accepted I can easily add it to the other extensions as well. It's just an added line in the access.xml file and an added ACL check in each view where the options button should appear.
### Testing
- Create a second user and adjust the permissions in com_content so he can access the options but doesn't have "Configure" permissions. 
- Check with SuperUser and this new user if the options work as expected. SuperUser should be able to save the permissions. The new user should not see them and not be able to save them (even if he fakes the form).
- Check also in other extensions that the SuperUser still can access and save the options and permissions. Nothing should have changed there.

Commit:
Add ACL `core.options` and take care of display the fieldset.

Commit:
Take care of saving

Commit:
Taking care of categories and featured view

Commit:
UI changes

sanderpotjer commented:
@Bakual was looking into the history of core.options, and found this PR. Was there any specific reason to not include core.options in the Global Configurations of Joomla? Right now a user can't allow the core.options action from a global level to allow the action for all components that are using that action.

Bakual commented:
@sanderpotjer I'd bet on it just being an oversight on my part. Because I added it first in com_content only, it didn't made sense to have it in the global options. And when I added it to the remaining extensions it just popped up in my mind to add it to global as well.

sanderpotjer commented:
@Bakual ok, thanks for the info! I will make a PR for this then.
